cmd/bucket-replication.go

```diff
@@ -117,6 +117,139 @@ func mustReplicater(ctx context.Context, r *http.Request, bucket, object string,
 	return cfg.Replicate(opts)
 }
 
+// returns true if any of the objects being deleted qualifies for replication.
+func mustReplicateDeletes(ctx context.Context, bucket string, objects []ObjectToDelete) bool {
+	c, err := getReplicationConfig(ctx, bucket)
+	if err != nil || c == nil {
+		return false
+	}
+	for _, obj := range objects {
+		if c.HasActiveRules(obj.ObjectName, true) {
+			return true
+		}
+	}
+	return false
+}
+
+// returns true if object qualifies for replication
+func isReplicateDelete(ctx context.Context, getObjectInfoFn GetObjectInfoFn, bucket string, dobj ObjectToDelete) bool {
+	rcfg, err := getReplicationConfig(ctx, bucket)
+	if err != nil || rcfg == nil {
+		return false
+	}
+	oi, err := getObjectInfoFn(ctx, bucket, dobj.ObjectName, ObjectOptions{VersionID: dobj.VersionID})
+	if err != nil {
+		validReplStatus := false
+		switch oi.ReplicationStatus {
```

Comment:
if err != nil, then oi will be invalid right?

Reply:
for delete markers, GetObjectInfo will return additional information - [see here]( https://github.com/minio/minio/pull/10396/files#diff-cdc839a13cf6812147035703a6129193R436)

Reply:
Should you check for `errFileNotFound`?

Reply:
I think you need to check explicitly for `errFileNotFound` as the reasons this calls fails are many and there is no guarantee on what is returned.